iree/compiler/Codegen/Common/InsertDistributionInfoPass.cpp

```diff
@@ -26,62 +26,31 @@ namespace iree_compiler {
 //===---------------------------------------------------------------------===//
 
-// Get the lowering configuration for the operation within the dispatch.
-// This looks for tile sizes by looking for lowering configuration.
-static FailureOr<SmallVector<int64_t>> getTileSizesFromLoweringConfig(
-    ArrayRef<Operation *> computeOps, MLIRContext *context) {
-  if (computeOps.empty()) return SmallVector<int64_t>{};
-
-  Optional<SmallVector<int64_t>> distributedTileSizes;
-  for (auto op : computeOps) {
-    auto partitionbleLoopInterface =
-        dyn_cast<IREE::Flow::PartitionableLoopsInterface>(op);
-    if (!partitionbleLoopInterface) continue;
-    IREE::Codegen::LoweringConfigAttr currLoweringConfig =
-        getLoweringConfig(op);
-    if (!currLoweringConfig) continue;
-    SmallVector<unsigned> partitionableLoops =
-        partitionbleLoopInterface.getPartitionableLoops(kNumMaxParallelDims);
-    SmallVector<int64_t> tileSizes = currLoweringConfig.getTileSizeVals(0);
-    SmallVector<int64_t> currDistributedTileSizes;
-    if (!partitionableLoops.empty()) {
-      currDistributedTileSizes.resize(partitionableLoops.back() + 1, 0);
-    }
-    for (auto loopID : partitionableLoops) {
-      if (loopID < tileSizes.size()) {
-        currDistributedTileSizes[loopID] = tileSizes[loopID];
-      }
-    }
-    if (distributedTileSizes) {
-      if (currDistributedTileSizes != distributedTileSizes) {
-        // Inconsistent distributed tile sizes. Abort.
-        return static_cast<LogicalResult>(
-            computeOps.front()->emitOpError("inconsistent distribution of ops "
-                                            "for first level of distribution"));
-      }
-    } else {
-      distributedTileSizes = currDistributedTileSizes;
-    }
-  }
-  if (distributedTileSizes) {
-    return distributedTileSizes.getValue();
-  }
-  return SmallVector<int64_t>{};
-}
-
 /// Compute the workload per workgroup to use based on the tile sizes passed.
 static SmallVector<int64_t> getWorkloadPerWorkgroup(
-    ArrayRef<int64_t> distributedLoopTileSizes) {
+    ArrayRef<int64_t> tileSizes, ArrayRef<int64_t> interchange) {
   // TODO(ravishankarm): This for now assumes that we can just drop all the
   // zero-dim tile sizes. We need to eventually change this so that we dont have
   // to do this. It is implicity linked to the dispatch region workload having
   // the consistent information. That needs to be changed to take the entire
   // iteration domain size as the argument, and then we can use the distribute
   // loop tile sizes directly.
+  SmallVector<int64_t> interchangedTileSizes;
+  if (interchange.empty()) {
+    interchangedTileSizes.assign(tileSizes.begin(), tileSizes.end());
+  } else {
+    interchangedTileSizes.resize(tileSizes.size());
+    for (auto en : llvm::enumerate(interchange)) {
+      // The check is needed because only the distributable dims are set.
+      if (en.value() < tileSizes.size()) {
```

Comment:
Not guaranteed that only the innermost dimensions are reductions. You need to look at partitionable loop interface here to get the partitioned loops.

Something I would like to fix is instead of constantly invoking the partitionable loops interface, make `partitionable_loops` also part of the lowering config. Then you have all the information you need here.